no-literal-checker/src/main/java/org/checkerframework/checker/noliteral/NoLiteralAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,191 @@
+package org.checkerframework.checker.noliteral;
+
+import java.util.List;
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.Element;
+import javax.lang.model.type.TypeKind;
+import javax.lang.model.type.TypeMirror;
+import javax.lang.model.util.Elements;
+import org.checkerframework.checker.noliteral.qual.MaybeDerivedFromConstant;
+import org.checkerframework.checker.noliteral.qual.NonConstant;
+import org.checkerframework.common.basetype.BaseAnnotatedTypeFactory;
+import org.checkerframework.common.basetype.BaseTypeChecker;
+import org.checkerframework.framework.qual.TypeUseLocation;
+import org.checkerframework.framework.type.AnnotatedTypeFactory;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+import org.checkerframework.framework.type.AnnotatedTypeMirror.AnnotatedArrayType;
+import org.checkerframework.framework.type.AnnotatedTypeMirror.AnnotatedDeclaredType;
+import org.checkerframework.framework.type.typeannotator.ListTypeAnnotator;
+import org.checkerframework.framework.type.typeannotator.TypeAnnotator;
+import org.checkerframework.framework.util.defaults.QualifierDefaults;
+import org.checkerframework.javacutil.AnnotationBuilder;
+import org.checkerframework.javacutil.ElementUtils;
+import org.checkerframework.javacutil.TypesUtils;
+
+/** The type factory for the no literal checker. */
+public class NoLiteralAnnotatedTypeFactory extends BaseAnnotatedTypeFactory {
+
+  /** The canonical maybe-constant annotation */
+  private final AnnotationMirror MAYBE_CONSTANT =
+      AnnotationBuilder.fromClass(elements, MaybeDerivedFromConstant.class);
+
+  /** The canonical non-constant annotation */
+  private final AnnotationMirror NON_CONSTANT =
+      AnnotationBuilder.fromClass(elements, NonConstant.class);
+
+  public NoLiteralAnnotatedTypeFactory(BaseTypeChecker checker) {
+    super(checker);
+    this.postInit();
+  }
+
+  /** @return a canonical non-constant AnnotationMirror */
+  public AnnotationMirror getNonConstant() {
+    return NON_CONSTANT;
+  }
+
+  /** @return a canonical maybe-constant AnnotationMirror */
+  public AnnotationMirror getMaybeConstant() {
+    return MAYBE_CONSTANT;
+  }
+
+  @Override
+  protected TypeAnnotator createTypeAnnotator() {
```

Comment:
Yes, it is needed, both for subclasses of `TypeAnnotator` and of `TreeAnnotator`. See for example [the Interning Checker](https://github.com/typetools/checker-framework/blob/master/checker/src/main/java/org/checkerframework/checker/interning/InterningAnnotatedTypeFactory.java#L124) or [the Lower Bound Checker](https://github.com/typetools/checker-framework/blob/master/checker/src/main/java/org/checkerframework/checker/index/lowerbound/LowerBoundAnnotatedTypeFactory.java#L233). It is a pain that these are needed, but it's been true for a long time and was never an issue before. I certainly don't object to you opening a bug report, but I have no interest in shepherding one or fixing it, because I don't think it's that inconvenient.